model-engine/model_engine_server/inference/tensorrt-llm/triton_model_repo/postprocessing/1/model.py

```diff
@@ -147,10 +176,25 @@ def finalize(self):
         """
         print("Cleaning up...")
 
-    def _postprocessing(self, tokens_batch):
+    def _postprocessing(self, tokens_batch, sequence_lengths):
         outputs = []
-        for beam_tokens in tokens_batch:
-            for tokens in beam_tokens:
-                output = self.tokenizer.decode(tokens)
+        for batch_idx, beam_tokens in enumerate(tokens_batch):
+            for beam_idx, tokens in enumerate(beam_tokens):
+                seq_len = sequence_lengths[batch_idx][beam_idx]
+                output = self.tokenizer.decode(
+                    tokens[:seq_len], skip_special_tokens=self.skip_special_tokens
+                )
+                # Adapted from https://github.com/triton-inference-server/tensorrtllm_backend/pull/423
+                # This is somewhat of a hack: add a space before the output if the first token starts with a space
+                # This may add a space in front of the first token though when we don't want it.
+                token_id_string = self.tokenizer.convert_ids_to_tokens(
```

Comment:
this is expensive though? we're effectively decoding twice. if we're only checking string for first token, can we just do ```
self.tokenizer.convert_ids_to_tokens(
                    tokens[0], skip_special_tokens=self.skip_special_tokens
                )```?